From 75ae827f8ef9f5ebb15e6a78b7745b168e28f50b Mon Sep 17 00:00:00 2001
From: Sijie Lin <sijielin@microsoft.com>
Date: Thu, 3 Nov 2022 15:11:02 -0700
Subject: [PATCH 01/11] Add test logs

---
 libfreerdp/core/freerdp.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/libfreerdp/core/freerdp.c b/libfreerdp/core/freerdp.c
index 7d9eba428..4b495006f 100644
--- a/libfreerdp/core/freerdp.c
+++ b/libfreerdp/core/freerdp.c
@@ -209,12 +209,19 @@ BOOL freerdp_connect(freerdp* instance)
 		goto freerdp_connect_finally;
 	}
 
+	WLog_INFO(TAG, "instance->settings->DumpRemoteFx: %" PRId32 "", instance->settings->DumpRemoteFx);
+	WLog_INFO(TAG, "instance->settings->DumpRemoteFxFile: %s", instance->settings->DumpRemoteFxFile);
+
 	if (instance->settings->DumpRemoteFx)
 	{
+		WLog_INFO(TAG, "instance->settings->DumpRemoteFx is True");
 		instance->update->pcap_rfx = pcap_open(instance->settings->DumpRemoteFxFile, TRUE);
 
 		if (instance->update->pcap_rfx)
+		{
+			WLog_INFO(TAG, "instance->update->pcap_rfx for DumpRemoteFxFile not empty, pcap_open succeeded");
 			instance->update->dump_rfx = TRUE;
+		}	
 	}
 
 	if (status)
@@ -244,8 +251,13 @@ BOOL freerdp_connect(freerdp* instance)
 		goto freerdp_connect_finally;
 	}
 
+	WLog_INFO(TAG, "instance->settings->PlayRemoteFx: %" PRId32 "", instance->settings->PlayRemoteFx);
+	WLog_INFO(TAG, "instance->settings->PlayRemoteFxFile: %s", instance->settings->PlayRemoteFxFile);
+
 	if (instance->settings->PlayRemoteFx)
 	{
+		WLog_INFO(TAG, "instance->settings->PlayRemoteFx is True");
+
 		wStream* s;
 		rdpUpdate* update;
 		pcap_record record;
@@ -256,7 +268,10 @@ BOOL freerdp_connect(freerdp* instance)
 		if (!update->pcap_rfx)
 			goto freerdp_connect_finally;
 		else
+		{
+			WLog_INFO(TAG, "instance->update->pcap_rfx for PlayRemoteFxFile not empty, pcap_open succeeded");
 			update->play_rfx = TRUE;
+		}
 
 		status = TRUE;
 
-- 
2.24.1.windows.2

