From 188beb9b7cdf47e8820ab3e40b43199c5635166c Mon Sep 17 00:00:00 2001
From: Sijie Lin <sijielin@microsoft.com>
Date: Fri, 4 Nov 2022 12:05:53 -0700
Subject: [PATCH 02/11] Add more logs

---
 libfreerdp/core/fastpath.c |  2 ++
 libfreerdp/utils/pcap.c    | 10 ++++++++++
 2 files changed, 12 insertions(+)

diff --git a/libfreerdp/core/fastpath.c b/libfreerdp/core/fastpath.c
index 7a67638fe..845b0deb9 100644
--- a/libfreerdp/core/fastpath.c
+++ b/libfreerdp/core/fastpath.c
@@ -381,6 +381,8 @@ static int fastpath_recv_update(rdpFastPath* fastpath, BYTE updateCode, wStream*
 	          fastpath_update_to_string(updateCode), updateCode, Stream_GetRemainingLength(s));
 #endif
 
+	WLog_INFO(TAG, "!!!updateCode: %" PRId32 "", updateCode);
+
 	switch (updateCode)
 	{
 		case FASTPATH_UPDATETYPE_ORDERS:
diff --git a/libfreerdp/utils/pcap.c b/libfreerdp/utils/pcap.c
index 520f49e22..aa3ca96d2 100644
--- a/libfreerdp/utils/pcap.c
+++ b/libfreerdp/utils/pcap.c
@@ -181,7 +181,10 @@ rdpPcap* pcap_open(char* name, BOOL write)
 	pcap->fp = winpr_fopen(name, write ? "w+b" : "rb");
 
 	if (pcap->fp == NULL)
+	{
+		WLog_INFO(TAG, "pcap->fp == NULL");
 		goto fail;
+	}
 
 	if (write)
 	{
@@ -193,7 +196,11 @@ rdpPcap* pcap_open(char* name, BOOL write)
 		pcap->header.snaplen = 0xFFFFFFFF;
 		pcap->header.network = 0;
 		if (!pcap_write_header(pcap, &pcap->header))
+		{
+			WLog_INFO(TAG, "!pcap_write_header(pcap, &pcap->header)");
 			goto fail;
+		}
+			
 	}
 	else
 	{
@@ -201,7 +208,10 @@ rdpPcap* pcap_open(char* name, BOOL write)
 		pcap->file_size = _ftelli64(pcap->fp);
 		_fseeki64(pcap->fp, 0, SEEK_SET);
 		if (!pcap_read_header(pcap, &pcap->header))
+		{
+			WLog_INFO(TAG, "!pcap_read_header(pcap, &pcap->header)");
 			goto fail;
+		}
 	}
 
 	return pcap;
-- 
2.24.1.windows.2

