From ec48161954d0e7587d2fd3ae58ee968bd86065b3 Mon Sep 17 00:00:00 2001
From: Sijie Lin <sijielin@microsoft.com>
Date: Fri, 4 Nov 2022 12:42:03 -0700
Subject: [PATCH 03/11] Test recording feature

---
 libfreerdp/core/fastpath.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/libfreerdp/core/fastpath.c b/libfreerdp/core/fastpath.c
index 845b0deb9..ef03a92dc 100644
--- a/libfreerdp/core/fastpath.c
+++ b/libfreerdp/core/fastpath.c
@@ -507,7 +507,9 @@ static int fastpath_recv_update_data(rdpFastPath* fastpath, wStream* s)
 	BYTE compression;
 	BYTE compressionFlags;
 	UINT32 DstSize = 0;
+	size_t start;
 	BYTE* pDstData = NULL;
+	BYTE* mark;
 	rdpTransport* transport;
 
 	if (!fastpath || !s)
@@ -524,6 +526,9 @@ static int fastpath_recv_update_data(rdpFastPath* fastpath, wStream* s)
 	if (!transport)
 		return -1;
 
+	start = Stream_GetPosition(s);
+	Stream_GetPointer(s, mark);
+
 	if (!fastpath_read_update_header(s, &updateCode, &fragmentation, &compression))
 		return -1;
 
@@ -574,6 +579,14 @@ static int fastpath_recv_update_data(rdpFastPath* fastpath, wStream* s)
 		Stream_SealLength(fastpath->updateData);
 		Stream_SetPosition(fastpath->updateData, 0);
 		status = fastpath_recv_update(fastpath, updateCode, fastpath->updateData);
+		if (fastpath->rdp->update->dump_rfx == TRUE && fastpath->rdp->update->pcap_rfx)
+		{
+			const size_t size = Stream_GetPosition(s) - start;
+			BOOL isPcapAddSucceeded = pcap_add_record(fastpath->rdp->update->pcap_rfx, mark, size);
+			WLog_INFO(TAG, "!!!!!Recording FASTPATH_FRAGMENT_SINGLE!!!!! size: %" PRId32 ", isPcapAddSucceeded: %" PRId32 "",
+			          size, isPcapAddSucceeded);
+			pcap_flush(fastpath->rdp->update->pcap_rfx);
+		}
 		Stream_SetPosition(fastpath->updateData, 0);
 
 		if (status < 0)
@@ -627,6 +640,16 @@ static int fastpath_recv_update_data(rdpFastPath* fastpath, wStream* s)
 			Stream_SealLength(fastpath->updateData);
 			Stream_SetPosition(fastpath->updateData, 0);
 			status = fastpath_recv_update(fastpath, updateCode, fastpath->updateData);
+			if (fastpath->rdp->update->dump_rfx == TRUE && fastpath->rdp->update->pcap_rfx)
+			{
+				const size_t size = Stream_GetPosition(s) - start;
+				BOOL isPcapAddSucceeded =
+				    pcap_add_record(fastpath->rdp->update->pcap_rfx, mark, size);
+				WLog_INFO(TAG,
+				          "!!!!!Recording FASTPATH_FRAGMENT_LAST!!!!! size: %" PRId32 ", isPcapAddSucceeded: %" PRId32 "",
+				          size, isPcapAddSucceeded);
+				pcap_flush(fastpath->rdp->update->pcap_rfx);
+			}
 			Stream_SetPosition(fastpath->updateData, 0);
 
 			if (status < 0)
-- 
2.24.1.windows.2

