From d9e11fd05fb646690bea35d2d839f8d91af88659 Mon Sep 17 00:00:00 2001
From: Sijie Lin <sijielin@microsoft.com>
Date: Wed, 9 Nov 2022 15:09:35 -0800
Subject: [PATCH 10/11] more logs in transport.c

---
 libfreerdp/core/transport.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/libfreerdp/core/transport.c b/libfreerdp/core/transport.c
index c0d1a0939..4151519eb 100644
--- a/libfreerdp/core/transport.c
+++ b/libfreerdp/core/transport.c
@@ -763,6 +763,8 @@ int transport_read_pdu(rdpTransport* transport, wStream* s)
 			else
 				pduLength = header[1];
 
+			WLog_INFO(TAG, "pduLength: %d", pduLength);
+
 			/*
 			 * fast-path has 7 bits for length so the maximum size, including headers is 0x8000
 			 * The theoretical minimum fast-path PDU consists only of two header bytes plus one
@@ -1000,6 +1002,7 @@ int transport_drain_output_buffer(rdpTransport* transport)
 
 int transport_check_fds(rdpTransport* transport)
 {
+	WLog_INFO(TAG, "=== In transport_check_fds ===");
 	int status;
 	int recv_status;
 	wStream* received;
@@ -1040,6 +1043,9 @@ int transport_check_fds(rdpTransport* transport)
 		 * Note that transport->ReceiveBuffer is replaced after each iteration
 		 * of this loop with a fresh stream instance from a pool.
 		 */
+		BYTE* buffer1 = Stream_Buffer(transport->ReceiveBuffer);
+		WLog_INFO(TAG, "Before transport_read_pdu: transport->ReceiveBuffer is %02X%02X%02X%02X",
+		          buffer1[0], buffer1[1], buffer1[2], buffer1[3]);
 		if ((status = transport_read_pdu(transport, transport->ReceiveBuffer)) <= 0)
 		{
 			if (status < 0)
@@ -1054,6 +1060,10 @@ int transport_check_fds(rdpTransport* transport)
 		if (!(transport->ReceiveBuffer = StreamPool_Take(transport->ReceivePool, 0)))
 			return -1;
 
+		BYTE* buffer2 = Stream_Buffer(received);
+		WLog_INFO(TAG, "Before ReceiveCallback: received is %02X%02X%02X%02X",
+		          buffer2[0], buffer2[1], buffer2[2], buffer2[3]);
+
 		/**
 		 * status:
 		 * 	-1: error
-- 
2.24.1.windows.2

